src/pcre2test.c

```diff
@@ -7368,27 +7389,37 @@ while ((c = *p++) != 0)
         }
       *q8++ = (uint8_t)c;
       }
+    else
+      {
+      if (c > 0x7fffffff)
+        {
+        fprintf(outfile, "** Character \\N{U+%x} is greater than 0x7fffffff "
+                         "and therefore cannot be encoded as UTF-8\n", c);
```

Comment:
`0x10ffff` is the highest unicode character regardless of encoding.

Reply:
Correct, but pcre2test allows encoding up to this original 6 byte maximum with the other escapes, and indeed Perl goes even further.

It becomes handy as  way to input very high values for the 32bit library too, which is why it is also allowed even in non utf mode.

Reply:
True, but pcre2test uses the original UTF-8 encoding scheme (RFC 2279) so as to handle characters up to `0x7fffffff.`

Reply:
noticed that Perl now [recommends](https://perldoc.perl.org/perlunicode#Comparison-of-%5CN%7B...%7D-and-%5Cp%7Bname=...%7D) using the `\p{name=` property for that which might come out handy, ex: `\p{name=/(SMILING|GRINNING) FACE/}`

Reply:
hopefully addressed in the followup